In [ ]:
!pip install pyspark
!pip install -U -q PyDrive
!apt install openjdk-8-jdk-headless -qq
import os
os.environ["JAVA_HOME"] = "/usr/lib/jvm/java-8-openjdk-amd64"

     |████████████████████████████████| 204.2MB 58kB/s 
     |████████████████████████████████| 204kB 27.7MB/s 
  Created wheel for pyspark: filename=pyspark-3.0.1-py2.py3-none-any.whl size=204612242 sha256=070efe02fdf926eec071b17a5e1f07621f2532760a53779767edb59eaee860c7
  Stored in directory: /root/.cache/pip/wheels/5e/bd/07/031766ca628adec8435bb40f0bd83bb676ce65ff4007f8e73f
Successfully built pyspark
The following additional packages will be installed:
  openjdk-8-jre-headless
Suggested packages:
  openjdk-8-demo openjdk-8-source libnss-mdns fonts-dejavu-extra
  fonts-ipafont-gothic fonts-ipafont-mincho fonts-wqy-microhei
  fonts-wqy-zenhei fonts-indic
The following NEW packages will be installed:
  openjdk-8-jdk-headless openjdk-8-jre-headless
0 upgraded, 2 newly installed, 0 to remove and 15 not upgraded.
Need to get 36.5 MB of archives.
After this operation, 143 MB of additional disk space will be used.
Selecting previously unselected package openjdk-8-jre-headless:amd64.
(Reading

In [ ]:
from pydrive.auth import GoogleAuth
from pydrive.drive import GoogleDrive
from google.colab import auth
from oauth2client.client import GoogleCredentials

auth.authenticate_user()
gauth = GoogleAuth()
gauth.credentials = GoogleCredentials.get_application_default()
drive = GoogleDrive(gauth)

In [ ]:
id='1f_9EbnywCj35EBUA32sueigxBjBJwALr'
downloaded = drive.CreateFile({'id': id})
downloaded.GetContentFile('War and Peace by Leo Tolstoy (ru).txt')

In [ ]:
import pandas as pd

import pyspark
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark import SparkContext, SparkConf

In [ ]:
conf = SparkConf().set("spark.ui.port", "4050")

sc = pyspark.SparkContext(conf=conf)
spark = SparkSession.builder.getOrCreate()

In [ ]:
spark

In [ ]:
!wget https://bin.equinox.io/c/4VmDzA7iaHb/ngrok-stable-linux-amd64.zip
!unzip ngrok-stable-linux-amd64.zip
get_ipython().system_raw('./ngrok http 4050 &')
!curl -s http://localhost:4040/api/tunnels | python3 -c \
    "import sys, json; print(json.load(sys.stdin)['tunnels'][0]['public_url'])"

--2020-12-23 12:45:24--  https://bin.equinox.io/c/4VmDzA7iaHb/ngrok-stable-linux-amd64.zip
Resolving bin.equinox.io (bin.equinox.io)... 34.232.47.250, 52.2.56.23, 35.153.56.97, ...
Connecting to bin.equinox.io (bin.equinox.io)|34.232.47.250|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 13773305 (13M) [application/octet-stream]
Saving to: ‘ngrok-stable-linux-amd64.zip.1’

ngrok-stable-linux- 100%[===================>]  13.13M  55.0MB/s    in 0.2s    

2020-12-23 12:45:24 (55.0 MB/s) - ‘ngrok-stable-linux-amd64.zip.1’ saved [13773305/13773305]

Archive:  ngrok-stable-linux-amd64.zip
replace ngrok? [y]es, [n]o, [A]ll, [N]one, [r]ename: y
  inflating: ngrok                   
http://d160ab737e6b.ngrok.io


Подсчитай кол-во слов в документе "War and Peace by Leo Tolstoy (ru).txt"

In [ ]:
# код
with open('War and Peace by Leo Tolstoy (ru).txt', 'rb') as f:
  text = f.read().decode("cp1251").encode("utf-8").decode("utf-8")
  new_file = open("new_file.txt", "w")
  new_file.write(text)
  new_file.close()

In [ ]:
chars_to_remove = ",.–"
def remove_in_string(new_string):
  string = new_string
  for character in chars_to_remove:
    string = string.replace(character, "")
  return string

In [ ]:
with open("new_file.txt", "r") as input, open("without_delimeters_last.txt", "w") as output:
  lines = input.readlines()
  for line in lines:
    print(line)
    line_without = remove_in_string(line)
    print(line_without)
    try:
      output.write(line_without)
    except Exception:
      pass

Streaming output truncated to the last 5000 lines.
 Тафа-лафа а что бормочет ничего не разберешь  говорил солдат передразнивая отъехавшего генерала  Расстрелял бы я их подлецов!





– В девятом часу велено на месте быть, а мы и половины не прошли. Вот так распоряжения! – повторялось с разных сторон.

 В девятом часу велено на месте быть а мы и половины не прошли Вот так распоряжения!  повторялось с разных сторон





И чувство энергии, с которым выступали в дело войска, начало обращаться в досаду и в злобу на бестолковые распоряжения и на немцев.

И чувство энергии с которым выступали в дело войска начало обращаться в досаду и в злобу на бестолковые распоряжения и на немцев





Причина путаницы заключалась в том, что во время движения австрийской кавалерии, шедшей на левом фланге, высшее начальство нашло, что наш центр слишком отдален от правого фланга, и всей кавалерии велено было перейти на правую сторону. Несколько тысяч кавалерии продвигалось перед пехотой, и пехота должна была ж

IOPub data rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_data_rate_limit`.

Current values:
NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)
NotebookApp.rate_limit_window=3.0 (secs)



Streaming output truncated to the last 5000 lines.
 Ну Петр Кириллович пойдем мы тебя отведем





В совершенной темноте солдаты вместе с Пьером пошли к Можайску.

В совершенной темноте солдаты вместе с Пьером пошли к Можайску





Уже петухи пели, когда они дошли до Можайска и стали подниматься на крутую городскую гору. Пьер шел вместе с солдатами, совершенно забыв, что его постоялый двор был внизу под горою и что он уже прошел его. Он бы не вспомнил этого (в таком он находился состоянии потерянности), ежели бы с ним не столкнулся на половине горы его берейтор, ходивший его отыскивать по городу и возвращавшийся назад к своему постоялому двору. Берейтор узнал Пьера по его шляпе, белевшей в темноте.

Уже петухи пели когда они дошли до Можайска и стали подниматься на крутую городскую гору Пьер шел вместе с солдатами совершенно забыв что его постоялый двор был внизу под горою и что он уже прошел его Он бы не вспомнил этого (в таком он находился состоянии потерянности) ежели бы с ним не ст

IOPub data rate exceeded.
The notebook server will temporarily stop sending output
to the client in order to avoid crashing it.
To change this limit, set the config variable
`--NotebookApp.iopub_data_rate_limit`.

Current values:
NotebookApp.iopub_data_rate_limit=1000000.0 (bytes/sec)
NotebookApp.rate_limit_window=3.0 (secs)



Streaming output truncated to the last 5000 lines.
– Сюда, сюда!

 Сюда сюда!

















287

287









Погодите, я сейчас сойду.

Погодите я сейчас сойду

















288

288









Эй, вы, живее, припекать начинает.

Эй вы живее припекать начинает

















289

289









Вот ваш ребенок. А, девочка, тем лучше. До свидания, толстяк. Что ж, надо по человечеству. Все люди.

Вот ваш ребенок А девочка тем лучше До свидания толстяк Что ж надо по человечеству Все люди

















290

290









Оставьте эту женщину!

Оставьте эту женщину!

















291

291









Ну, ну! Не дури!

Ну ну! Не дури!

















292

292









Поручик, у него кинжал.

Поручик у него кинжал

















293

293









А, оружие!

А оружие!

















294

294









Хорошо, хорошо, на суде все расскажешь.

Хорошо хорошо на суде все расскажешь

















295

295









Говоришь ли по-французски?

Говоришь ли по-французски?
















In [ ]:
from pyspark.sql import SparkSession
ss = SparkSession(sc)

In [ ]:
df = ss.read.csv(path="without_delimeters_last.txt", lineSep=' ')

In [ ]:
df.count()

460344

То есть в романе 460344 слова. Это достаточно близко к данным из интернета (~478 000 слов)

Дальше сделать уже просто не успеваю:(

In [ ]:
id='13yfAoONwq4rS5XrTv3IrcqcFcdgfvK9V'
downloaded = drive.CreateFile({'id': id})
downloaded.GetContentFile('mnist-digits-train.txt')

id='1VE_9x0LQvOJpHXbXp_RMPl3Q4wRUuOok'
downloaded = drive.CreateFile({'id': id})
downloaded.GetContentFile('mnist-digits-test.txt')

Необходимо обучить модель используя Spark MLlib (модель на ваш выбор, например Decision Tree) и получить accuracy.
Подробнее тут: https://spark.apache.org/docs/latest/ml-classification-regression.html

In [ ]:
# код

In [ ]:
id='1kUIrskM0zNH8u71G9M1BkHjRQYxvgAvh'
downloaded = drive.CreateFile({'id': id})
downloaded.GetContentFile('data.zip')
!unzip data.zip

Кто получает больше всего? Кто меньше всего?

In [ ]:
# код

Выведете топ 5 по зарплате.

In [ ]:
# код

Сколько всего регионов? Сколько работников в каждом регионе?

In [ ]:
# код

Выведете всех работников из Китая.

In [ ]:
# код

Укажите самую высокооплачиваемою должность.

In [ ]:
# код

Выведете всех работников связанных с ИТ. Выведете их менеджеров. 

In [ ]:
# код

Выведете имя и фамилию работника, его текущую и предыдущую должности и сколько полных недель и дней прошло с момент изменения. 



In [ ]:
# код

Выведете уникальные телефонные номера

In [ ]:
# код

Есть ли сотрудники с одинаковыми фамилиями и сколько их.

In [ ]:
# код